packages/cli/src/lib/transpile/morph/functions.ts

```diff
@@ -20,31 +21,58 @@ export function generateFunctionDocumentation(
 
   generateParameterDocumentation(functionNode, outputDocNode);
 
-  // Add type annotation to exported functions
-  // using the `export const foo: Foo = () => {}` syntax
-  const jsDocs = variableStatement?.getJsDocs()[0];
-  if (jsDocs?.getTags().length === 0) {
+  if (variableStatement) {
+    // Add type annotation to exported functions
+    // using the `export const foo: Foo = () => {}` syntax
     const declaration = variableStatement?.getDeclarations()[0];
-    const type = declaration?.getType().getText();
-    if (type && type !== 'any' && type !== 'unknown') {
-      const tagName = type.startsWith('() =>') ? 'return' : 'type';
-
-      // Some generated docs try to import variables from the same file.
-      // E.g. `<typeof loader>` becomes `<typeof import("....").loader>`
-      const normalizedType = type
-        .replace(/^\(\)\s+=>\s+/, '')
-        .replace(/typeof import\("[^"]+"\)\./, 'typeof ');
-
-      const text =
-        normalizedType === 'SerializeFrom<typeof loader>'
-          ? `{LoaderReturnData}` // Better alias
-          : `{${normalizedType}}`;
-
-      jsDocs.addTag({tagName, text});
+    const jsDocs = variableStatement?.getJsDocs()[0];
+
+    if (jsDocs?.getTags().length === 0) {
+      const type = declaration?.getType().getText();
+
+      if (isAnnotableType(type)) {
+        jsDocs.addTag({
+          tagName: type.startsWith('() =>') ? 'return' : 'type',
+          text: normalizeType(type),
+        });
+      }
+    }
+  } else {
+    // Add return type annotation to plain functions
+    const declaration = functionNode as FunctionDeclaration;
+    const jsDocs = declaration.getJsDocs()[0];
+
+    if (jsDocs?.getTags().length === 0) {
+      const returnType = declaration.getReturnType().getText();
+      if (isAnnotableType(returnType)) {
+        jsDocs.addTag({tagName: 'return', text: normalizeType(returnType)});
+      }
     }
   }
```

Comment:
This new part is adding `@return` on top of exported functions like `export function useRootLoaderData()....`.

The original part was only working for `export const useRootLoaderData = () =>...` syntax.